code/game/objects/structures/barsign.dm

```diff
@@ -59,22 +59,27 @@
 /obj/structure/sign/barsign/attack_ai(mob/user as mob)
 	return src.attack_hand(user)
 
+/obj/structure/sign/barsign/attackby(var/obj/item/I, var/mob/user)
+		if(istype(I, /obj/item/stack/cable_coil) && panel_open)
+		var/obj/item/stack/cable_coil/C = I
+		if(emagged) //Emagged, not broken by EMP
+			to_chat(user, "<span class='warning'>Sign has been damaged beyond repair!</span>")
+			return
+		else if(!broken)
+			to_chat(user, "<span class='warning'>This sign is functioning properly!</span>")
+			return
 
+		if(C.use(2))
+			to_chat(user, "<span class='notice'>You replace the burnt wiring.</span>")
+			broken = 0
+		else
+			to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+	else
+		return ..()
 
-/obj/structure/sign/barsign/attack_hand(mob/user as mob)
-	if(!src.allowed(user))
-		to_chat(user, "<span class = 'info'>Access denied.</span>")
-		return
-	if(broken)
-		to_chat(user, "<span class ='danger'>The controls seem unresponsive.</span>")
-		return
-	pick_sign()
-
-
-
-
-/obj/structure/sign/barsign/attackby(var/obj/item/I, var/mob/user)
-	if( istype(I, /obj/item/screwdriver))
+// Added to prevent people from unscrewing the barsign, making go invisible in the process
+/obj/structure/sign/barsign/screwdriver_act()
+    if( istype(I, /obj/item/screwdriver))
```

Comment:
This check should also not exist. `screwdriver_act` means an item was used with the property of being a screwdriver tool. The old way used istype checks